Quaver.Shared/Screens/Gameplay/CustomAudioSampleCache.cs

```diff
@@ -0,0 +1,104 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * Copyright (c) Swan & The Quaver Team <support@quavergame.com>.
+*/
+
+using System.Collections.Generic;
+using Quaver.Shared.Database.Maps;
+using Wobble.Audio.Samples;
+
+namespace Quaver.Shared.Screens.Gameplay
+{
+    public class CustomAudioSampleCache
+    {
+        /// <summary>
+        ///     MD5 hash of the map where the sound samples are from.
+        /// </summary>
+        private string MapMd5 { get; set; }
+
+        /// <summary>
+        ///     The cached audio samples.
+        /// </summary>
+        private List<AudioSample> Samples { get; set; } = new List<AudioSample>();
+
+        /// <summary>
+        ///     Currently playing channels.
+        /// </summary>
+        private List<AudioSampleChannel> Channels { get; set; } = new List<AudioSampleChannel>();
+
+        /// <summary>
+        ///     Loads audio samples for the specified map into the cache.
+        /// </summary>
+        /// <param name="map"></param>
+        public void LoadSamples(Map map)
+        {
+            // Always clean up the left-over channels.
+            StopAll();
+
+            // If the MD5 is the same, no need to re-load the samples.
+            if (MapMd5 == map.Md5Checksum)
```

Comment:
`map.Md5Checksum` will use the md5 that is cached in the database. Use `GameplayScreen.MapHash` if you want to retrieve the most up-to-date one.

Reply:
What about when it's called from `EditorScreen`?

Reply:
You can use the cached version. Doesn't truly matter in this case since it gets updated upon saving.